akka-sample-sharding-scala/src/main/scala/sample/sharding/Device.scala

```diff
@@ -1,54 +1,61 @@
 package sample.sharding
 
-import akka.actor._
+import akka.actor.typed.{ ActorRef, Behavior }
+import akka.actor.typed.scaladsl.Behaviors
 
 /**
-  * This is just an example: cluster sharding would be overkill for just keeping a small amount of data,
-  * but becomes useful when you have a collection of 'heavy' actors (in terms of processing or state)
-  * so you need to distribute them across several nodes.
-  */
+ * This is just an example: cluster sharding would be overkill for just keeping a small amount of data,
+ * but becomes useful when you have a collection of 'heavy' actors (in terms of processing or state)
+ * so you need to distribute them across several nodes.
+ */
 object Device {
+
   sealed trait Command extends Message
 
-  case class RecordTemperature(deviceId: Int, temperature: Double)
-      extends Command
+  final case class RecordTemperature(deviceId: Int, temperature: Double) extends Command
 
-  case class GetTemperature(deviceId: Int) extends Command
+  final case class GetTemperature(deviceId: Int, replyTo: ActorRef[Message]) extends Command
 
-  case class Temperature(deviceId: Int,
-                         average: Double,
-                         latest: Double,
-                         readings: Int)
-      extends Message
+  final case class Temperature(deviceId: Int, average: Double, latest: Double, readings: Int) extends Message {
+    override def toString: String =
+      s"Temperature[device=$deviceId, temperature=$latest, average=$average, readings=$readings]"
```

Comment:
can't we use default toString?

Reply:
We can, though it seemed nice to not have each log statement do the work. Reverted.

Reply:
Also: overloaded method value info ...